### ISSUE TITLE
Dynamic slice with `uint8` index returns incorrect result

### ISSUE THREAD

**Issue by jakevdp** [openxla/xla#4491]
Reported in https://github.com/google/jax/issues/16836

Here's a minimal repro:
```python
import jax.numpy as jnp
from jax import lax

arr = jnp.arange(200)
i = jnp.uint8(128)
print(lax.dynamic_slice(arr, [i], [1]))
# [199]
```
I've double checked that this is not hitting any index normalization on the JAX side; I believe the input values are being passed more-or-less directly to the HLO here: https://github.com/google/jax/blob/b11661696cc7005816359a70242886efa785b429/jax/_src/interpreters/mlir.py#L1472-L1473

**Comment by jakevdp** [comment]
I realized that this issue has already been reported in https://github.com/google/jax/issues/6122, with an internal tracking bug at b/183216273

I'll leave this open because it seems to have fallen off the radar.

**Comment by hawkinsp** [comment]
https://github.com/openxla/xla/commit/4e4eff35bf9a5f8ed54fd290391bd0612f49533e fixed this.
